# Awesome Human BCR repertoire papers üìù

B cell receptor (BCR) repertoire analysis has grown tremendously in the last 10
years or so, and has been especially relevant for understanding disease and
finding novel therapeutics. We at [Alchemab](https://www.alchemab.com/) love
the huge growth in this space and have curated a list of papers across several
areas. This is not an exhaustive list, so please let us know if we've missed
your paper.

Where applicable, we also comment on data availability in the
[Observed Antibody Space (OAS) database](http://opig.stats.ox.ac.uk/webapps/oas/oas).

If you'd like to contribute, or provide any new categories, feel free to submit a pull request.

## Table of contents
* [Reviews](#reviews)
* [BCR repertoire datasets](#bcr-repertoire-datasets)
* [Clonotyping Methods](#clonotyping-methods)
* [Machine learning](#machine-learning)

### Reviews

* Pertseva _et al._, [Applications of Machine and Deep Learning in Adaptive Immunity](https://www.annualreviews.org/doi/abs/10.1146/annurev-chembioeng-101420-125021). _Annu Rev Chem Biomol Eng_ (2021).
* Raybould _et al._, [Current strategies for detecting functional convergence across B-cell receptor repertoires](https://www.tandfonline.com/doi/full/10.1080/19420862.2021.1996732). _mAbs_ (2021).
* Snapkov _et al._, [Progress and challenges in mass spectrometry-based analysis of antibody
  repertoires](https://linkinghub.elsevier.com/retrieve/pii/S0167-7799(21)00193-1). _Trends Biotechnol_ (2021).
  
* Cspregi _et al._, [Immune Literacy: Reading, Writing, and Editing Adaptive Immunity](https://linkinghub.elsevier.com/retrieve/pii/S2589-0042(20)30711-2).
  _iScience_ (2020).
* Greiff _et al._, [Mining adaptive immune receptor repertoires for biological and clinical information 
  using machine learning](https://www.sciencedirect.com/science/article/pii/S2452310020300524). _Curr Opin Sys Biol_ (2020).
* Marks _et al._, [How repertoire data are changing antibody science](https://www.jbc.org/article/S0021-9258(17)48926-5).
  _J Biol Chem_ (2020).  
* Rees, [Understanding the human antibody repertoire](https://www.tandfonline.com/doi/full/10.1080/19420862.2020.1729683).
  _mAbs_, (2020).
* Teraguchi _et al._, [Methods for sequence and structural analysis of B and T cell receptor repertoires](https://www.sciencedirect.com/science/article/pii/S2001037020303408). _Comput Struct Biotechnol J_ (2020).

* Brown _et al._, [Augmenting adaptive immunity: progress and challenges in the quantitative engineering and analysis of adaptive immune receptor
  repertoires](https://pubs.rsc.org/en/content/articlehtml/2019/me/c9me00071b). _Mol Syst Des Eng_ (2019).
* Davis and Boyd, [Recent progress in the analysis of Œ±Œ≤T cell and B cell receptor repertoires](https://www.sciencedirect.com/science/article/abs/pii/S0952791519300482).
  _Curr Opin Immunol_ (2019).

* Bashford-Rogers _et al._, [Antibody repertoire analysis in polygenic autoimmune diseases](https://onlinelibrary.wiley.com/doi/10.1111/imm.12927).
  _Immunology_ (2018).
* Chaudhary and Wesemann, [Analyzing Immunoglobulin Repertoires](https://www.frontiersin.org/articles/10.3389/fimmu.2018.00462/full).
_Front Immunol_ (2018).
* Collins and Watson, [Immunoglobulin Light Chain Gene Rearrangements, Receptor Editing and the Development of a
  Self-Tolerant Antibody Repertoire](https://www.frontiersin.org/articles/10.3389/fimmu.2018.02249/full). _Front Immunol_ (2018).
* Ghraichy _et al._, [B-cell receptor repertoire sequencing in patients with primary immunodeficiency: a
  review](https://onlinelibrary.wiley.com/doi/10.1111/imm.12865). _Immunology_ (2018).  
* Miho _et al._, [Computational Strategies for Dissecting the High-Dimensional Complexity of Adaptive Immune
  Repertoires](https://www.frontiersin.org/articles/10.3389/fimmu.2018.00224/full). _Front Immunol_ (2018).
* Parola _et al._, [Integrating high-throughput screening and sequencing for monoclonal antibody discovery and
  engineering](https://onlinelibrary.wiley.com/doi/10.1111/imm.12838) _Immunology_ (2018).  
* Yermanos _et al._, [Tracing Antibody Repertoire Evolution by Systems Phylogeny](https://www.frontiersin.org/articles/10.3389/fimmu.2018.02149/full).
  _Front Immunol_ (2018).
  
* Friedensohn _et al._, [Advanced Methodologies in High-Throughput Sequencing of Immune
  Repertoires](https://linkinghub.elsevier.com/retrieve/pii/S0167-7799(16)30162-7). _Trends Biotechnol_ (2017).  
* Kovaltsuk _et al._, [How B-Cell Receptor Repertoire Sequencing Can Be Enriched with Structural Antibody Data
](https://www.frontiersin.org/articles/10.3389/fimmu.2017.01753/full). _Front Immunol_ (2017).
  
* Yaari and Kleinstein, [Practical guidelines for B-cell receptor repertoire sequencing
  analysis](https://genomemedicine.biomedcentral.com/articles/10.1186/s13073-015-0243-2). _Genome Med_ (2015).
* Boyd _et al._, [Deep sequencing and human antibody repertoire analysis](https://www.sciencedirect.com/science/article/abs/pii/S0952791516300255).
  _Curr Opin Immunol_ (2016).
  
* Greiff _et al._, [Bioinformatic and Statistical Analysis of Adaptive Immune Repertoires
](https://linkinghub.elsevier.com/retrieve/pii/S1471-4906(15)00223-9). _Trends Immunol_ (2015).
* Hershberg and Luning Prak, [The analysis of clonal expansions in normal and autoimmune B cell
  repertoires](https://royalsocietypublishing.org/doi/full/10.1098/rstb.2014.0239). _Philos Trans R Soc Lond B Biol Sci_
  (2015).
* Kearney _et al._, [Natural antibody repertoires: development and functional role in inhibiting allergic airway
  disease](https://www.annualreviews.org/doi/10.1146/annurev-immunol-032713-120140). _Annu Rev Immunol_ (2015).
* Lavinder _et al._, [Next-generation sequencing and protein mass spectrometry for the comprehensive analysis of human
  cellular and serum antibody repertoires](https://www.sciencedirect.com/science/article/abs/pii/S1367593114001719).
  _Curr Opin Chem Biol_ (2015).
* Robinson, [Sequencing the functional antibody repertoire--diagnostic and therapeutic discovery](https://www.nature.com/articles/nrrheum.2014.220).
  _Nat Rev Rheumatol_ (2015).

* Galson _et al._, [Studying the antibody repertoire after vaccination: practical
  applications](https://doi.org/10.1016/j.it.2014.04.005). _Trends Immunol_ (2014).
* Georgiou _et al._, [The promise and challenge of high-throughput sequencing of the antibody repertoire
](https://www.nature.com/articles/nbt.2782). _Nat Biotechnol_ (2014).
  
* Six _et al._ [The past, present, and future of immune repertoire biology ‚Äì the rise of next-generation
  repertoire analysis](https://www.frontiersin.org/articles/10.3389/fimmu.2013.00413/full) _Front Immunol_ (2013).
* Reddy _et al._ [Systems analysis of adaptive immunity by utilization of high-throughput technologies](https://linkinghub.elsevier.com/retrieve/pii/S0958-1669(11)00078-4).
  _Curr Opin Biotechnol_ (2011).

### BCR repertoire datasets

#### Notes
We refer readers to the [Observed Antibody Space (OAS) database](http://opig.stats.ox.ac.uk/webapps/oas/oas) and the
[Paired OAS database](https://onlinelibrary.wiley.com/doi/10.1002/pro.4205) as the most straight-forward resources
for obtaining amino acid sequence datasets for _e.g._ machine learning applications. 

For simplicity, we're mentioning articles that are _not already covered by OAS_. We focus on papers where authors
have uploaded datasets as part of the publication (with no login/registration required) or via public repositories such
as the NIH SRA, requiring minimal admin overhead. Publications that only provide V gene + J gene + CDR3 are not mentioned
here. 

* Aizik _et al._ [Antibody Repertoire Analysis of Tumor-Infiltrating B Cells Reveals Distinct Signatures and Distributions
  Across Tissues](https://www.frontiersin.org/articles/10.3389/fimmu.2021.705381) _Front Immunol_ (2021).
* Ehling _et al._ [SARS-CoV-2 reactive and neutralizing antibodies discovered by single-cell sequencing of plasma cells
  and mammalian display](https://www.cell.com/cell-reports/fulltext/S2211-1247(21)01751-4). _Cell Rep_ (2021).
* Harris _et al._ [Tumor-Infiltrating B Lymphocyte Profiling Identifies IgG-Biased, Clonally Expanded Prognostic
  Phenotypes in Triple-Negative Breast Cancer](https://cancerres.aacrjournals.org/content/81/16/4290.long). _Caner Res_ (2021).
  **Data available directly from publication.**
* Ohm-Laursen _et al._ [B Cell Mobilization, Dissemination, Fine Tuning of Local Antigen Specificity and Isotype Selection
  in Asthma](https://www.frontiersin.org/articles/10.3389/fimmu.2021.702074/). _Front Immunol_ (2021).
* Turner _et al._ [SARS-CoV-2 mRNA vaccines induce persistent human germinal centre responses](https://www.nature.com/articles/s41586-021-03738-2).
  _Nature_ (2021). **Data available in SRA**.
* Sokal _et al._ [Maturation and persistence of the anti-SARS-CoV-2 memory B cell response](https://www.cell.com/cell/fulltext/S0092-8674(21)00093-3).
  _Cell_ (2021). **Data available in EBI ArrayExpress**.
* Stewart _et al._ [Single-Cell Transcriptomic Analyses Define Distinct Peripheral B Cell Subsets and Discrete Development
  Pathways](https://www.frontiersin.org/articles/10.3389/fimmu.2021.602539/full). **Data available in EBI ArrayExpress**.  
* Wieland _et al._ [Defining HPV-specific B cell responses in patients with head and neck cancer](https://www.nature.com/articles/s41586-020-2931-3). _Nature_ (2021).
  **Data available via GEO**.
* Yang _et al._ [Large-scale analysis of 2,152 Ig-seq datasets reveals key features of B cell biology and the antibody
  repertoire](https://www.cell.com/cell-reports/fulltext/S2211-1247(21)00444-7). _Cell Rep_ (2021).
* Galson _et al._ [Deep Sequencing of B Cell Receptor Repertoires From COVID-19 Patients Reveals Strong Convergent Immune
  Signatures](https://www.frontiersin.org/articles/10.3389/fimmu.2020.605170/full). _Front Immunol_ (2020). **Data available
  directly from publication.**
* Li _et al._ [Mucosal or systemic microbiota exposures shape the B cell repertoire](https://www.nature.com/articles/s41586-020-2564-6). 
  _Nature_ (2020). **Data in SRA.**
* Ramesh _et al._ [A pathogenic and clonally expanded B cell transcriptome in active multiple sclerosis](https://www.pnas.org/content/117/37/22932).
  _PNAS_ (2020). **Data in SRA and GEO.**
* Roskin _et al._ [Aberrant B cell repertoire selection associated with HIV neutralizing antibody breadth](https://www.nature.com/articles/s41590-019-0581-0).
  _Nat Immunol_ (2020). **Data in SRA.**
* Woodruff _et al._ [Extrafollicular B cell responses correlate with neutralizing antibodies and morbidity in COVID-19](https://www.nature.com/articles/s41590-020-00814-z).
_Nat Immunol_ (2020). **Data in SRA.**

### Machine learning
#### Notes
The remit of this section is to cover how machine learning (ML) methods have been trained _using_ BCR repertoire data,
_or_ applying existing ML tools for enhancing repertoire analyses. Due to the nature of the way things are moving in this
space, quite a few publications are just pre-prints!

* Kim _et al._ [Computational analysis of B cell receptor repertoires in COVID-19 patients using deep embedded
  representations of protein sequences](https://www.biorxiv.org/content/10.1101/2021.08.02.454701v3). _biorXiv_ (2021).
* Leem _et al._ [Deciphering the language of antibodies using self-supervised learning](https://www.biorxiv.org/content/10.1101/2021.11.10.468064v1).
_biorXiv_ (2021).
* Marks _et al._ [Humanization of antibodies using a machine learning approach on large-scale repertoire data](https://academic.oup.com/bioinformatics/article/37/22/4041/6295884).
  _Bioinformatics_ (2021).
* Ostrovsky-Berman _et al._ [Immune2vec: Embedding B/T Cell Receptor Sequences in ‚ÑùN Using Natural Language Processing](https://www.frontiersin.org/articles/10.3389/fimmu.2021.680687/full).
  _Front Immunol_ (2021).
* Pavlovic _et al._ [The immuneML ecosystem for machine learning analysis of adaptive immune receptor
  repertoires](https://www.nature.com/articles/s42256-021-00413-z). _Nat Mach Intell_ (2021).
* Prihoda _et al._ [BioPhi: A platform for antibody design, humanization and humanness evaluation based on natural
  antibody repertoires and deep learning](https://www.biorxiv.org/content/10.1101/2021.08.08.455394v1). _biorXiv_ (2021). 
* Richardson _et al._ [A computational method for immune repertoire mining that identifies novel binders from different
  clonotypes, demonstrated by identifying anti-pertussis toxoid antibodies](https://www.tandfonline.com/doi/full/10.1080/19420862.2020.1869406). _mAbs_ (2021).  
* Ruffolo _et al._ [Antibody structure prediction using interpretable deep learning](https://www.cell.com/patterns/fulltext/S2666-3899(21)00280-4).
  _Patterns_ (2021a).
* Ruffolo _et al._ [Deciphering antibody affinity maturation with language models and weakly supervised learning](https://arxiv.org/abs/2112.07782).
_arXiv_ (2021b).
* Shemesh _et al._ [Machine Learning Analysis of Na√Øve B-Cell Receptor Repertoires Stratifies Celiac Disease Patients
  and Controls](https://www.frontiersin.org/articles/10.3389/fimmu.2021.627813/full). _Front Immunol_ (2021).  
* Shuai _et al._ [Generative Language Modeling for Antibody Design](https://www.biorxiv.org/content/10.1101/2021.12.13.472419v1) _biorXiv_ (2021).
* Amimeur _et al._ [Designing Feature-Controlled Humanoid Antibody Discovery Libraries Using Generative Adversarial
  Networks](https://www.biorxiv.org/content/10.1101/2020.04.12.024844v2). _biorXiv_ (2020).
* Widrich _et al._ [Modern Hopfield Networks and Attention for Immune Repertoire Classification](). _NeurIPS Proceedings_
  (2020).  
* Wollacott _et al._ [Quantifying the nativeness of antibody sequences using long short-term memory networks](https://academic.oup.com/peds/article/32/7/347/5554642).
_Protein Eng Des Sel_ (2019).


### Clonotyping Methods

#### Notes
The remit of this section is to cover "methods papers" that focus on how to identify clonotypes from immune BCR repertoire
datasets.

* Lindenbaum _et al._, [Alignment free identification of clones in B cell receptor repertoires](https://academic.oup.com/nar/article/49/4/e21/6039915). 
  _Nucleic Acids Res_ (2021).  
* Nouri and Kleinstein, [A spectral clustering-based method for identifying clones from high-throughput B cell repertoire sequencing data](https://academic.oup.com/bioinformatics/article/34/13/i341/5045726).
  _Bioinformatics_ (2018).
* Gupta _et al._, [Hierarchical clustering can identify B cell clones with high confidence in Ig repertoire sequencing
  data](https://www.jimmunol.org/content/198/6/2489.long). _J Immunol_ (2017).